src/test_hash_table.py

```diff
@@ -0,0 +1,72 @@
+# _*_ encoding: utf-8 _*_
+import pytest
+import io
+
+
+
+@pytest.fixture()
+def hash_table_20():
+    """Fixture that makes a hash table with 20 buckets."""
+    from hash_table import HashTable
+    ht = HashTable(20)
+    return ht
+
+@pytest.fixture()
+def big_table():
+    """Fixture that makes a hash table with 20 buckets."""
+    from hash_table import HashTable
+    ht = HashTable(50)
+    return ht
+
+
+@pytest.fixture()
+def hash_table_2():
+    """Fixture that makes a hash table with 2 buckets."""
+    from hash_table import HashTable
+    ht = HashTable(2)
+    return ht
+
+
+def test_hash(hash_table_2):
+    """Ensure hashing algorithm works as expected."""
+    assert hash_table_2._hash('a') == 1
+
+
+def test_longer_hash(hash_table_20):
+    """Check a longer string is hashed as expected."""
+    assert hash_table_20._hash('abcdef') == 17
+
+
+def not_string(hash_table_20):
+    """Ensure TypeError raises when key is not string."""
+    with pytest.raises:
```

Comment:
If it's supposed to raise a TypeError then test for that specifically, not just any exception that could be thrown.